src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -2133,6 +2133,1855 @@ struct S
+
+        if (new C<string?>() is [ 1, ..var rest3 ])
+            rest3.ToString(); // 5
+        else
+            rest3.ToString(); // 6, 7
+
+        if (new C<string>() is [ 1, ..var rest4 ])
+        {
+            rest4.ToString();
+            rest4 = null;
+        }
+        else
+            rest4.ToString(); // 8, 9
+    }
+}
+";
+            var compilation = CreateCompilation(new[] { source, TestSources.Index, TestSources.Range });
+            compilation.VerifyEmitDiagnostics(
+                // (15,13): error CS0165: Use of unassigned local variable 'rest'
+                //             rest.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "rest").WithArguments("rest").WithLocation(15, 13),
+                // (18,13): warning CS8629: Nullable value type may be null.
+                //             rest2.Value.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "rest2").WithLocation(18, 13),
+                // (20,13): warning CS8629: Nullable value type may be null.
+                //             rest2.Value.ToString(); // 3, 4
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "rest2").WithLocation(20, 13),
+                // (20,13): error CS0165: Use of unassigned local variable 'rest2'
+                //             rest2.Value.ToString(); // 3, 4
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "rest2").WithArguments("rest2").WithLocation(20, 13),
+                // (23,13): warning CS8602: Dereference of a possibly null reference.
+                //             rest3.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "rest3").WithLocation(23, 13),
+                // (25,13): warning CS8602: Dereference of a possibly null reference.
+                //             rest3.ToString(); // 6, 7
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "rest3").WithLocation(25, 13),
+                // (25,13): error CS0165: Use of unassigned local variable 'rest3'
+                //             rest3.ToString(); // 6, 7
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "rest3").WithArguments("rest3").WithLocation(25, 13),
+                // (33,13): warning CS8602: Dereference of a possibly null reference.
+                //             rest4.ToString(); // 8, 9
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "rest4").WithLocation(33, 13),
+                // (33,13): error CS0165: Use of unassigned local variable 'rest4'
+                //             rest4.ToString(); // 8, 9
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "rest4").WithArguments("rest4").WithLocation(33, 13)
+                );
+        }
+
+        [Fact]
+        public void SlicePattern_Nullability_Array()
+        {
+            var source = @"
+#nullable enable
+class C
+{
+    public void M()
+    {
+        if (new int[0] is [ 1, ..var rest ])
```

Comment:
It's probably worth adding some tests that don't use a `var` pattern, but instead use a `not null` or similar pattern.